src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -698,20 +665,38 @@ private static bool HaveSameTypeConstraints(TypeParameterSymbol typeParameter1,
                 AreConstraintTypesSubset(substitutedTypes2, substitutedTypes1, typeParameter1);
         }
 
-        public static bool HaveSameNullabilityInConstraints(TypeParameterSymbol typeParameter1, TypeMap typeMap1, TypeParameterSymbol typeParameter2, TypeMap typeMap2)
+        public static bool HaveSameNullabilityInConstraints(TypeParameterSymbol typeParameter1, TypeMap? typeMap1, TypeParameterSymbol typeParameter2, TypeMap? typeMap2, TypeCompareKind typeComparison)
```

Comment:
>TypeCompareKind typeComparison [](http://example.com/codeflow?start=170&length=30)

Looking at this method more, I realized that it is probably not appropriate to let user control the comparison type used by this method because the supplied value might not be suitable for the task that the method is supposed to perform. I suggest reverting changes to this method and not using it for the purpose of this PR. Instead the top level nullability comparison can be done explicitly in ```HaveSameCSharpSignature``` helper. It is already doing the other checks by calling ```HaveSameConstraints```.